src/frontend/js/components/AshleyEditor/index.tsx

```diff
@@ -121,19 +132,55 @@ export const AshleyEditor = (props: MyEditorProps) => {
     return 'not-handled';
   };
 
+  let mentionPluginRender;
+  const plugins = [toolbarPlugin, emojiPlugin, linkPlugin, codeEditorPlugin];
+
+  if (props.mention_users) {
+    const [open, setOpen] = useState(true);
+    const [suggestions, setSuggestions] = useState(props.mention_users);
+
+    const { MentionSuggestions, mentionPlugin } = useMemo(() => {
+      // tslint:disable-next-line:no-shadowed-variable
+      const mentionPlugin = createMentionPlugin();
+      // tslint:disable-next-line:no-shadowed-variable
+      const { MentionSuggestions } = mentionPlugin;
+      return { mentionPlugin, MentionSuggestions };
+    }, []);
+
+    const onOpenChange = useCallback((_open: boolean) => {
+      setOpen(_open);
+    }, []);
+    const onSearchChange = useCallback(({ value }: { value: string }) => {
+      setSuggestions(defaultSuggestionsFilter(value, props.mention_users!));
+    }, []);
+    plugins.push(mentionPlugin);
+    mentionPluginRender = (
+      <MentionSuggestions
+        open={open}
+        onOpenChange={onOpenChange}
+        suggestions={suggestions}
+        onSearchChange={onSearchChange}
+        onAddMention={() => {
+          // get the mention object selected
+        }}
+      />
+    );
+  }
+
   return (
     <div className="ashley-editor-wrapper">
       <div className="ashley-editor-widget" onClick={focusEditor}>
         <Editor
           ref={editorRef}
           editorState={editorState}
           onChange={editorChange}
-          plugins={[toolbarPlugin, emojiPlugin, linkPlugin, codeEditorPlugin]}
+          plugins={plugins}
           placeholder={props.placeholder}
           handleKeyCommand={keyBinding}
           handlePastedText={handlePastedText}
         />
         <emojiPlugin.EmojiSuggestions />
+        {mentionPluginRender}
```

Comment:
As you did for plugins, why not create a `PluginRenderers` array with default `[emojiPlugin.EmojiSuggestions]`, then you could push `mentionPluginRender` ?

And in your component do a 
```jsx
PluginRenderers.map(Plugin => <Plugin />)
```